### DISCUSSION TITLE
Hands free stopping transcription

### DISCUSSION THREAD

**Discussion by Emaa719** [deepgram/community#723] [General help]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

i am working on a live streaming transcription using microphone, Originally, the user would end their speech by pressing ENTER. However, I want the system to be hands-free. This means that when the user stops talking, the transcription should be automatically returned, and  the function finishes.
this is the code 

"
async def record_transcribe() -> str:
    full_transcript = ""
    try:
        from deepgram import DeepgramClient, LiveTranscriptionEvents, LiveOptions, Microphone

        # Initialize Deepgram client
        deepgram: DeepgramClient = DeepgramClient(DEEPGRAM_API_KEY)
        dg_connection = deepgram.listen.live.v("1")

        full_transcript = ""

        def on_open(self, open, **kwargs):
            print(f"\n\n{open}\n\n")

        def on_message(self, result, **kwargs):
            sentence = result.channel.alternatives[0].transcript
            if len(sentence) != 0:
                nonlocal full_transcript
                full_transcript += sentence + " "
            print(f"speaker: {sentence}")

        def on_utterance_end(self, utterance_end, **kwargs):
            print(f"\n\nUtterance ended. Stopping transcription.\n\n")
            nonlocal full_transcript
            dg_connection.finish() 
            print("Full transcript:", full_transcript)

        def on_error(self, error, **kwargs):
            print(f"\n\n{error}\n\n")

        # Attach event listeners
        dg_connection.on(LiveTranscriptionEvents.Open, on_open)
        dg_connection.on(LiveTranscriptionEvents.Transcript, on_message)
        dg_connection.on(LiveTranscriptionEvents.UtteranceEnd, on_utterance_end)
        dg_connection.on(LiveTranscriptionEvents.Error, on_error)

        options: LiveOptions = LiveOptions(
            model="nova-2-general",
            punctuate=True,
            language="fr",
            encoding="linear16",
            channels=1,
            sample_rate=16000,
            # To get UtteranceEnd, the following must be set:
            interim_results=True,
            utterance_end_ms="1000",
            endpointing=5000
        )

        if dg_connection.start(options) is False:
            print("Failed to connect to Deepgram")
            return

        # Open a microphone stream on the default input device
        microphone = Microphone(dg_connection.send)

        # Start microphone
        microphone.start()

        # Attend la fermeture du microphone, déclenchée par l'événement UtteranceEnd
        microphone.finish()

        # Indicate that we've finished
        print("Finished")
        return full_transcript

    except Exception as e:
        print(f"Could not open socket: {e}")
        return """
        
        but i got this error 
        
        "{'type': 'exception', 'description': 'unknown error _listening', 'message': 'received 1011 (internal error) deepgram did not receive audio data or a text message within the timeout window. see https://dpgr.am/net0001}"

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment]
This seems reasonable, is it not working for you? I will only note that with UtteranceEnd, this would then finish and kill the connection even if the user is simply making a long-ish (~1-2 second) pause in their speech.

**Reply by Emaa719** [reply]
hello,
i got this error "    "{'type': 'exception', 'description': 'unknown error _listening', 'message': 'received 1011 (internal error) deepgram did not receive audio data or a text message within the timeout window. see https://dpgr.am/net0001}"
"

**Reply by nikolawhallon** [reply]
Do you received this error after having received transcripts, or before, or after utterance end fires? The message is saying you have stopped sending audio data to Deepgram, and I just want to know when that is occurring.

**Reply by Emaa719** [reply]
I launch the script, but my voice is not recorded by the microphone, once the function starts it displays listening and at the same time displays finished, afterward the error is thrown

print("Listening...")
transcription = await record_transcribe()
print("Done listening")
